contracts/RelayHub.sol

```diff
@@ -362,7 +363,7 @@ contract RelayHub is IRelayHub {
     }
 
     function calculateCharge(uint256 gasUsed, GsnTypes.RelayData calldata relayData) public override virtual view returns (uint256) {
-        return relayData.baseRelayFee + (gasUsed * relayData.gasPrice * (100 + relayData.pctRelayFee)) / 100;
+        return relayData.baseRelayFee.add((gasUsed.mul(relayData.gasPrice).mul(100 + relayData.pctRelayFee)).div(100));
```

Comment:
we're not protecting against any invalid values of the relayData. 
but instead of check we can limit them in range, as  none of these parameters require 256 bits.

pctRelayFee can safely use 32bit
baseRelayFee, gasPrice can fit <112bit
and thus they are protected against overflows, reduce bit-pollution (i.e reduce request size by 2 words)

Reply:
and even if not - they should be validated on registerRelay, not on each usage.

Reply:
A client can send a request with whatever value if the server agrees to it/the server can be the one manipulating with these values. Again, I don't think there is anything wrong with this code - but better safe then sorry

Reply:
And limiting uint size is something I really don't like as an idea

Reply:
a. you left *two* unchecked values: `pctRelayFee` and `baseRelayFee` (remember: `add(a,b)` only verify that `a+b > b`, and make no assumption about a)
b. if you change it do `add(100)`, the expression bcms cmpltly uncpmrhnsibl -
   all for the sake of making it "correct".
c. So a better way is validating the parameters (e.g. `require(pctGasPrice<1e10 && baseRelayFee<1e10)`) - but I still think that using a bounded types are better.
d. side-note: `div(100)` is useless, since it only verifies that `100>0`

Reply:
This is the wrong check, in the wrong place.
The method is called multiple times (any TokenPaymaster-style paymaster), so the params should be checked, 
liek other params, at the beginning of `relayCall()`.
We do check some params there, but neglect to check others.

should add a line there:

```
require(relayRequest.relayData.baseRelayFee<MAXINT && relayRequest.relayData.pctRelayFee < MAXINT, "Invalid fees");
```

(and outside: `uint constant MAXINT = uint96(-1);`)

note that no need to check other params of this function, since there is already check for `gasPrice` in relayCall, and `gasused` is also a bound value.